backend/globaleaks/utils/users_details_filter.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8
+import re
+
+class UserDetailsFilter:
+    """
+    Filters the text for any user details.
+    Removes things like uuid4, email, pgp formatted strings, session ids.
+    """
+
+    def __init__(self, text):
+        """
+        Filters or scrubs the given text.
+        Arguments:
+            text {string} -- text to be fileted.
+        """
+        self._text = text
+
+    def remove_bgp_formatted_string(self):
+        # remove public key block
+        self._text = re.sub(r'-----BEGIN PGP PUBLIC KEY BLOCK-----.*-----END PGP PUBLIC KEY BLOCK-----',
+            "*****", self._text, 0, re.DOTALL)
+
+        # remove private key block
+        self._text = re.sub(r'-----BEGIN PGP PRIVATE KEY BLOCK-----.*-----END PGP PRIVATE KEY BLOCK-----',
```

Comment:
I would probably make a single regexp to filter at once any PGP message so just looking for BEGIN PGP and END PGP. 

What do you think?

Reply:
Thanks for the great idea I've replaced by single regex to filter any pgp string.